### PR TITLE
Upgrading syncthing to v1.12.1

### PR DESCRIPTION
**Pull request by vraravam**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by Rylan12** [comment]
Thanks for the PR. It looks like you're using a commit in homebrew/core. [`syncthing`](https://github.com/Homebrew/homebrew-core/blob/e80e7c0e1d4cda4e56afb0167f177c5a3ec6576f/Formula/syncthing.rb#L4) is currently at version 1.13.0.

https://github.com/Homebrew/homebrew-core/blob/e80e7c0e1d4cda4e56afb0167f177c5a3ec6576f/Formula/syncthing.rb#L1-L4

**Comment by vraravam** [comment]
> Thanks for the PR. It looks like you're using an old commit in homebrew/core. [`syncthing`](https://github.com/Homebrew/homebrew-core/blob/e80e7c0e1d4cda4e56afb0167f177c5a3ec6576f/Formula/syncthing.rb#L4) is currently at version 1.13.0.
> 
> https://github.com/Homebrew/homebrew-core/blob/e80e7c0e1d4cda4e56afb0167f177c5a3ec6576f/Formula/syncthing.rb#L1-L4

@ryan-norton - as far as the syncthing website is concerned, the publicly released version is 1.12.1 - the one that I have referenced. The version 1.13.0 that you are referring to is an RC release.

**Comment by vraravam** [comment]
Sorry - could you please reopen this PR @Rylan12 ?

**Comment by Rylan12** [comment]
Ah, you're right. Can you rebase this PR onto the latest master commit?

You'll also need to add a `version_scheme` to revert the version. Let me know if you need any help with either of those.

**Comment by vraravam** [comment]
@Rylan12 - both are done - please review.

File: Formula/syncthing.rb [hunk 1]
```diff
@@ -1,14 +1,17 @@
 class Syncthing < Formula
   desc "Open source continuous file synchronization application"
   homepage "https://syncthing.net/"
-  url "https://github.com/syncthing/syncthing/archive/v1.13.0.tar.gz"
-  sha256 "b31d4a323c6c970c580ecab3fd2327c5ba2709286b766b157139a5e393795bc2"
+  url "https://github.com/syncthing/syncthing.git",
+      tag:      "v1.12.1",
+      revision: "c1b452df9365e0857fa3e9f9d229ead65cd1ed5d"
+
```
Comment (Rylan12) [inline Formula/syncthing.rb:7]:
```suggestion
  url "https://github.com/syncthing/syncthing/archive/v1.12.1.tar.gz"
  sha256 "5751365374e03c32f731aeb1d07f7df0984eca30e14878d636b91cd35f0db1c3"
```

Is there a reason to change to the git url? I think we'd rather stay with the archive URL if possible.

**Comment by carlocab** [comment]
Looks like syncthing 1.13 is expected today: https://forum.syncthing.net/

Maybe we can wait and see if they come through on that, and merge this if not?

**Comment by Rylan12** [comment]
Good find. I agree, I'd rather avoid changing the version scheme for a downgrade just to be reverted (but keeping the new version scheme) in just a few hours.

File: Formula/syncthing.rb (hunk 1 above)
Comment (vraravam) [reply Formula/syncthing.rb]: I have reverted to the archive URL in the new commit.

**Comment by carlocab** [comment]
1.13 was released two hours ago: https://github.com/syncthing/syncthing/releases/tag/v1.13.0

**Comment by carlocab** [comment]
I'll just fix the livecheck block in a separate PR. Thanks for opening this, @vraravam!

**Pull request closed without merging**